unstructured_ingest/v2/processes/connectors/lancedb.py

```diff
@@ -0,0 +1,161 @@
+import asyncio
+import json
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Optional
+
+import pandas as pd
+from pydantic import Field, Secret
+
+from unstructured_ingest.error import DestinationConnectionError
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces.connector import AccessConfig, ConnectionConfig
+from unstructured_ingest.v2.interfaces.file_data import FileData
+from unstructured_ingest.v2.interfaces.upload_stager import UploadStager, UploadStagerConfig
+from unstructured_ingest.v2.interfaces.uploader import Uploader, UploaderConfig
+from unstructured_ingest.v2.processes.connector_registry import DestinationRegistryEntry
+
+CONNECTOR_TYPE = "lancedb"
+
+if TYPE_CHECKING:
+    from lancedb import AsyncConnection
+
+
+SUPPORTED_ELEMENT_FIELDS = (
+    "text",
+    "element_id",
+    "type",
+)
+SUPPORTED_ELEMENT_METADATA_FIELDS = (
+    "filename",
+    "is_continuation",
+    "file_type",
+    "page_number",
+    "text_as_html",
+)
+
+
+class LanceDBAccessConfig(AccessConfig):
+    aws_access_key_id: Optional[str] = Field(
+        default=None, description="The AWS access key ID to use."
+    )
+    aws_secret_access_key: Optional[str] = Field(
+        default=None, description="The AWS secret access key to use."
+    )
+    google_service_account_key: Optional[str] = Field(
+        default=None, description="The serialized google service account key."
+    )
+    azure_storage_account_name: Optional[str] = Field(
+        default=None, description="The name of the azure storage account."
+    )
+    azure_storage_account_key: Optional[str] = Field(
+        default=None, description="The serialized azure service account key."
+    )
+
+    @property
+    def storage_options(self) -> dict:
+        storage_options = {
+            "aws_access_key_id": self.aws_access_key_id,
+            "aws_secret_access_key": self.aws_secret_access_key,
+            "google_service_account_key": self.google_service_account_key,
+            "azure_storage_account_name": self.azure_storage_account_name,
+            "azure_storage_account_key": self.azure_storage_account_key,
+        }
+        return {key: value for key, value in storage_options.items() if value is not None}
+
+
+class LanceDBConnectionConfig(ConnectionConfig):
+    access_config: Secret[LanceDBAccessConfig] = Field(
+        default_factory=LanceDBAccessConfig, validate_default=True
+    )
+    uri: str = Field(description="The uri of the database.")
+
+    @requires_dependencies(["lancedb"], extras="lancedb")
+    @DestinationConnectionError.wrap
+    async def get_async_connection(self) -> "AsyncConnection":
+        import lancedb
+
+        return await lancedb.connect_async(
```

Comment:
Since the `AsyncConnection` uses a context manager, can we actually wrap this method in one so that the connection returned is closed by default? Take a look at how the sql connectors expose a connection using this approach: [snowflake.py](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/v2/processes/connectors/sql/snowflake.py#L63)